main/src/main/java/SimpleNotePad.java

```diff
@@ -0,0 +1,344 @@
+import javax.swing.JTextPane;
+import javax.swing.text.Position;
+import javax.swing.text.StyledDocument;
+
+public class SimpleNotePad extends JFrame{
+    JMenuBar uiMenuBar = new JMenuBar();
+    JMenu uiMenuFile = new JMenu("File");
+    JMenu uiMenuEdit = new JMenu("Edit");
+    JMenu uiMenuRecent = new JMenu("Recent");
+    JTextPane uiTextPane = new JTextPane();
+    JMenuItem uiMenuNew = new JMenuItem("New File");
+    JMenuItem uiMenuSave = new JMenuItem("Save File");
+    JMenuItem uiMenuPrint = new JMenuItem("Print File");
+    JMenuItem uiMenuCopy = new JMenuItem("Copy");
+    JMenuItem uiMenuPaste = new JMenuItem("Paste");
+    JMenuItem uiMenuOpen = new JMenuItem("Open");
+    JMenuItem uiMenuReplace = new JMenuItem("Replace");
+
+    Properties appProperties;
+
+    // Notepad Initialization
+    public SimpleNotePad(Properties configuration) {
+        this.setTitle("A Simple Notepad Tool");
+
+        appProperties = configuration;
+        String config = appProperties.getProperty("recent");
+        // Load recent files from config
+        if (config != null && config.length() > 0) {
+
+            String[] settings = config.split(",");
+            for (String newItem : settings) {
+                JMenuItem recentItem = new JMenuItem(newItem);
+                recentItem.addActionListener(new ActionListener() {
+                    public void actionPerformed(ActionEvent e) {
+                        try {
+                            JMenuItem sourceItem = (JMenuItem)e.getSource();
+                            FileReader file = new FileReader(sourceItem.getText());
+                            String inputText = "";
+                            int i;
+                            while ((i=file.read()) != -1) {
+                                inputText += (char)i;
+                            }
+                            uiTextPane.setText(inputText);
+                            JOptionPane.showMessageDialog(null, "File loading complete");
+                        } catch (Exception ex) {
+                            JOptionPane.showMessageDialog(null,
+                                    "Load error " + ex, "Load error",
+                                    JOptionPane.ERROR_MESSAGE);
+                        }
+                    }
+                });
+
+                uiMenuRecent.add(recentItem);
+            }
+        }
+
+        uiMenuFile.add(uiMenuOpen);
+        uiMenuFile.addSeparator();
+        uiMenuFile.add(uiMenuNew);
+        uiMenuFile.addSeparator();
+        uiMenuFile.add(uiMenuSave);
+        uiMenuFile.addSeparator();
+        uiMenuFile.add(uiMenuPrint);
+        uiMenuFile.addSeparator();
+        uiMenuFile.add(uiMenuRecent);
+
+        uiMenuEdit.add(uiMenuPaste);
+        uiMenuEdit.addSeparator();
+        uiMenuEdit.add(uiMenuCopy);
+        uiMenuEdit.addSeparator();
+        uiMenuEdit.add(uiMenuReplace);
+
+        uiMenuOpen.addActionListener(this.actionMenuOpen());
```

Comment:
Using lambda, you actually don't need to use the this.actionMenuOpen(). 

Just do uiMenuOpen.addActionListener(e -> {logic});